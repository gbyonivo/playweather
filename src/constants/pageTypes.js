export default {
  forecast: 'forecast',
  current: 'current'
};