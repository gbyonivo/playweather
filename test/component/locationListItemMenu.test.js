import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import LocationListMenu from '../../src/components/locationListMenu';
import combinedReducer from '../../src/reducers';
import pageTypes from '../../src/constants/pageTypes';

const props = {
  fetchData: () => { }
};

describe('LocationListMenu component without test', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: { locations: ['machetser, uk'] } });
    const actual = renderer.create(
      <MemoryRouter initialEntries={[{ pathname: '/', match: { params: { type: pageTypes.current } } }]}>
        <Provider store={store}>
          <LocationListMenu {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});