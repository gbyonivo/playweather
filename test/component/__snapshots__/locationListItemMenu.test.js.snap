// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationListMenu component without test renders correctly 1`] = `
<div
  className="locationListMenu"
>
  <h2>
    Recently Viewed
  </h2>
  <ul>
    <li
      className="locationListMenuItem"
      onClick={[Function]}
    >
      machetser, uk
    </li>
  </ul>
</div>
`;
